File: lib_common/src/main/java/com/yyxnb/common/log/LogConfig.java
package com.yyxnb.common.log;

import android.text.TextUtils;

import com.yyxnb.common.AppConfig;


public class LogConfig {

    private boolean showThreadInfo = true;
    private boolean debug = AppConfig.getInstance().isDebug();
    private String tag = "----arch----";


    public LogConfig setTag(String tag) {
        if (!TextUtils.isEmpty(tag)) {
            this.tag = tag;
        }
        return this;
    }

    public LogConfig setShowThreadInfo(boolean showThreadInfo) {
        this.showThreadInfo = showThreadInfo;
        return this;
    }


    public LogConfig setDebug(boolean debug) {
        this.debug = debug;
        return this;
    }

    public String getTag() {
        return tag;
    }

    public boolean isDebug() {
        return debug;
    }

    public boolean isShowThreadInfo() {
        return showThreadInfo;
    }
}

File: lib_common/src/main/java/com/yyxnb/common/utils/log/LogConfig.kt
package com.yyxnb.common.utils.log

import android.text.TextUtils
import com.yyxnb.common.CommonManager

object LogConfig {

    var isShowThreadInfo = true

    var isDebug = CommonManager.isDebug

    var tag = "----what----"

    fun setTag(tag: String): LogConfig {
        if (!TextUtils.isEmpty(tag)) {
            this.tag = tag
        }
        return this
    }

    fun setShowThreadInfo(showThreadInfo: Boolean): LogConfig {
        isShowThreadInfo = showThreadInfo
        return this
    }

    fun setDebug(debug: Boolean): LogConfig {
        this.isDebug = debug
        return this
    }

}